internal/rbd/encryption.go

```diff
@@ -341,7 +342,8 @@ func ParseEncryptionOpts(
 		encrypted, kmsID string
 	)
 	encrypted, ok = volOptions["encrypted"]
-	if !ok {
+	val, _ := strconv.ParseBool(encrypted)
```

Comment:
I'm not sure why parsing is necessary when the encrypted key isn't configured

Reply:
No need to parse in case `!ok`, but it is possible that `encrypted: false` is set. Checking the boolean value of `encrypted` is important.

Maybe you prefer something like this?

```go
encrypted, ok = volOptions["encrypted"]
if !ok {
    return ....
} else if val, _ := strconv.ParseBool(encrypted): !val {
    return ...
}
```

I am not in favor of two the same `return` statement though. Converting the `encrypted` string to a boolean, even when not required, is not a huge performance loss, and I think the code will be cleaner.

Reply:
We will need to follow the 2nd, so as to handle the case when the encrypted is not configured.